commit b070ced387a670b4fe1440f5e6ae086fe6e5ef3b
Author: James Antill <james@and.org>
Date:   Thu Oct 13 10:35:05 2011 -0400

    Ignore time skew by using sqlite IDs instead to order transactions. BZ 745635.

diff --git a/yum/history.py b/yum/history.py
index 8e62f50..d27fa44 100644
--- a/yum/history.py
+++ b/yum/history.py
@@ -1204,7 +1204,11 @@ class YumHistory:
         if tids and len(tids) <= yum.constants.PATTERNS_INDEXED_MAX:
             params = tids = list(set(tids))
             sql += " WHERE tid IN (%s)" % ", ".join(['?'] * len(tids))
-        sql += " ORDER BY beg_ts DESC, tid ASC"
+        #  This relies on the fact that the PRIMARY KEY in sqlite will always
+        # increase with each transaction. In theory we can use:
+        # ORDER BY beg_ts DESC ... except sometimes people do installs with a
+        # system clock that is very broken, and using that screws them forever.
+        sql += " ORDER BY tid DESC"
         if limit is not None:
             sql += " LIMIT " + str(limit)
         executeSQL(cur, sql, params)
